src/mcp_agent/workflows/llm/augmented_llm_openai.py

```diff
@@ -136,19 +157,27 @@ async def generate(self, message, request_params: RequestParams | None = None):
                 "messages": messages,
                 "stop": params.stopSequences,
                 "tools": available_tools,
-                "max_tokens": params.maxTokens,
             }
-
-            if available_tools:
-                arguments["tools"] = available_tools
-                arguments["parallel_tool_calls"] = params.parallel_tool_calls
+            if self._reasoning:
+                arguments = {
+                    **arguments,
+                    "max_completion_tokens": params.maxTokens,
+                    "reasoning_effort": self._reasoning_effort,
+                }
+            else:
+                arguments = {**arguments, "max_tokens": params.maxTokens}
+                if available_tools:
+                    arguments["parallel_tool_calls"] = params.parallel_tool_calls
```

Comment:
Should this be outside the if self._reasoning/else block?
```suggestion
            if available_tools:
                arguments["parallel_tool_calls"] = params.parallel_tool_calls
```

Reply:
it should; o3 models 400 if you send this parameter.